agent/internal/docker.go

```diff
@@ -127,6 +127,17 @@ func (d *dockerActor) pullImage(ctx *actor.Context, msg pullImage) {
 			sendErr(ctx, errors.Wrap(err, "error encoding registry credentials"))
 			return
 		}
+	} else if store, ok := newCredentialStore(ctx, reference.Domain(ref)); ok {
```

Comment:
we probably could "cache" the stores. or like, read the config at startup and see what all helpers we have and key them all on the domain they're for.